crates/modelardb_server/src/configuration.rs

```diff
@@ -31,7 +31,7 @@ use crate::ClusterMode;
 #[derive(Clone)]
 pub struct ConfigurationManager {
     /// Folder for storing metadata and Apache Parquet files on the local file system.
-    pub(crate) local_data_folder: PathBuf,
+    pub(crate) local_data_folder: Arc<LocalFileSystem>,
```

Comment:
I think it makes sense to keep `Arc` for struct fields since each struct needs to own a reference to the local data folder. As far as i can see it seems to be the best way to share the single local data folder across all components that need to use it.